src/bsoncxx/document/view.cpp

```diff
@@ -124,7 +125,8 @@ view::const_iterator view::const_iterator::operator++(int) {
 }
 
 bool operator==(const view::const_iterator& lhs, const view::const_iterator& rhs) {
-    return (lhs._element.raw == rhs._element.raw && lhs._element.offset == rhs._element.offset);
+    return (lhs._element.raw() == rhs._element.raw() &&
```

Comment:
Ah... I wonder if this is a job for http://en.cppreference.com/w/cpp/utility/tuple/forward_as_tuple